### PR TITLE
Crop

### PR DESCRIPTION
**Pull request by railsagainstignorance**

### CODE REVIEW

File: bin/lib/aws.js [hunk 1]
```diff
@@ -22,23 +22,44 @@ function init() {
...
+	return new Promise( resolve => {
+		setTimeout( () => {
+			resolve(millis);
+		}, millis);
+	});
+}
+
 async function translate(options) {
-	const text = await Utils.checkTextLength(options.text, BYTE_LIMIT);
+	var textChunks = await Utils.splitTextIntoChunks(options.text, BYTE_LIMIT);
```
Comment (Lily2point0) [inline bin/lib/aws.js]: `let`

File: bin/lib/aws.js [hunk 2]
```diff
@@ -22,23 +22,44 @@ function init() {
...
 		TargetLanguageCode: options.to.toLowerCase()
 	};
 
-	for (let i = 0; i < text.length; ++i) {
-		params.Text = text[i];
+	for (let i = 0; i < textChunks.length; ++i) {
+
+		if (i>0) { // pause before 2nd (and each subsequent) chunk
+			console.log(`AWS.translate: eventId=${options.translationEventId}, waiting ${THROTTLE_LIMIT} millis before submitting chunk=${i+1} (of ${textChunks.length}).`);
+			await pauseForMillis( THROTTLE_LIMIT );
```
Comment (Lily2point0) [inline bin/lib/aws.js]: why `await` and not a standard Timeout/Interval? Seems overkill.

File: index.js [hunk 3]
```diff
@@ -27,6 +27,8 @@ app.post('/article/:uuid/:lang', (req,res) => {
 	const uuid = req.params.uuid;
 	const lang = req.params.lang;
 	const translators = req.body.translators;
+	// default is firstChunkOnly=true
+	const firstChunkOnly = (!req.query.hasOwnProperty('firstChunkOnly')) || Boolean(req.query.firstChunkOnly);
```
Comment (Lily2point0) [inline index.js]: Missing `new` in front of `Boolean`, another way to enforce Boolean value is to use a double bang `!!`

File: bin/lib/aws.js [hunk 4]
```diff
@@ -22,23 +22,44 @@ function init() {
...
+		}, millis);
+	});
+}
+
 async function translate(options) {
-	const text = await Utils.checkTextLength(options.text, BYTE_LIMIT);
+	var textChunks = await Utils.splitTextIntoChunks(options.text, BYTE_LIMIT);
+	if (options.firstChunkOnly && textChunks.length > 1) {
+		console.log(`AWS.translate: eventId=${options.translationEventId}, firstChunkOnly, so discarding ${textChunks.length -1} (of ${textChunks.length}) chunks`);
+		textChunks = textChunks.slice(0,1);
```
Comment (Lily2point0) [inline bin/lib/aws.js]: `textChunks = textChunks[0];`

**Comment by railsagainstignorance** [comment]
updated. moved pauseForMillis into utils

**Pull request merged**